src/client/src/commons/form/borehole/borehole/geometry.jsx

```diff
@@ -8,21 +8,35 @@ import { FullPageCentered } from "../../../../components/baseComponents";
 import { useBoreholeGeometry, useBoreholeGeometryMutations } from "../../../../api/fetchApiV2";
 import { DeleteButton } from "../../../../components/buttons/buttons";
 
-const Geometry = ({ boreholeId, isEditable }) => {
+/**
+ * Renders the Geometry panel.
+ *
+ * @component
+ * @param {Object} props - The component props.
+ * @param {string} props.boreholeId - The ID of the borehole.
+ * @param {boolean} props.isEditable - Indicates whether the component is editable.
+ * @param {number} props.measuredDepth - The measured depth to show a perfectly vertical geometry if no geometry is set.
+ * @returns {JSX.Element} The rendered Geometry component.
+ */
+const Geometry = ({ boreholeId, isEditable, measuredDepth }) => {
   const { t } = useTranslation();
-
   const { data } = useBoreholeGeometry(boreholeId);
   const {
-    delete: { mutate: deleteBoreholeGeometry },
+    delete: { mutate: deleteBoreholeGeometry, isLoading: isDeletingBoreholeGeometry },
   } = useBoreholeGeometryMutations();
 
+  const defaultData = [
+    { md: 0, x: 0, y: 0, z: 0 },
+    { md: measuredDepth, x: 0, y: 0, z: measuredDepth },
+  ];
+
   return (
     <>
       {!data ? (
         <FullPageCentered>
           <CircularProgress />
         </FullPageCentered>
-      ) : data?.length === 0 && !isEditable ? (
```

Comment:
PP: ich finde den flow des conditional rendering hier in der Komponente etwas schwer nachzuvollziehen, vielleicht könnten Subkomponenten im gleichen File mit sprechenden Namen (zb. Charts, Actions o.ä.) und Konstanten wie "hasData" helfen ?

Reply:
Es wird unterschieden zwischen keine daten geladen `!data` und leere daten `data.length === 0`. Zudem macht es keinen Sinn die `defaultData` anzuzeigen, wenn im Borehole noch keine `measuredDepth` gesetzt wurden.

Subkomponenten finde ich in diesem Fall kontraproduktiv, da es die information was angezeigt wird weiter verstreut und verschachtelt. Bei den Actions ist es nämlich sehr relevant, das da ein Delete Button ist.

Ich habe die conditions mit Konstanten hoffentlich einfacher verständlich gemacht.